README.md

```diff
@@ -4,39 +4,41 @@
 
 # matrix-registration
 
-a simple python application to have a token based matrix registration
+A simple Python application enabling token-based registration for matrix servers.
 
-if you like me encountered the situation where you want to invite your friends to your homeserver, but neither wanted to open up public registration nor create accounts for every individual user yourself, this project should be the solution.
+You may have, like me, encountered the situation where you want to invite your friends to create an account on your homeserver, but neither want to open up public registration nor create accounts for every individual user yourself. This project aims to solve this problem.
 
-with this project you can just quickly generate tokens on the fly and share them with your friends to allow them to register to your homeserver.
+With matrix-registration, you can quickly generate tokens on the fly and share them with your friends to allow them to register on your homeserver.
 
 <img src="https://matrix.org/_matrix/media/v1/download/dmnd.sh/UKGgpbHRdFXzKywxjjbfHAsI" width="500">
 
 
 ## setup
+Install using pip:
 
 ```bash
 pip3 install matrix-registration
 matrix-registration
 ```
+
+### First start
+Generate a configuration for your matrix homeserver by copying [config.sample.yaml](/config.sample.yaml) to your server and editing it:
+```bash
+wget https://raw.githubusercontent.com/ZerataX/matrix-registration/master/config.sample.yaml
+cp config.sample.yaml config.yaml
+nano config.yaml
+```
+
+Then pass the path to this configuration to the application on startup using `--confiig-path /path/to/config.yaml`.
+
 __INFO:__ 
 - This only asks you for the most important options. 
 You should definitely take a look at the actual configuration file.
 - The `shared_secret` has to be the same as `registration_shared_secret` in your homeserver.yaml
 
-### nginx reverse-proxy
```

Comment:
I assume a reverse proxy (tho maybe not nginx) is part of everyone's setup here, so I don't see why it should be moved?

the project itself doesn't support TLS and using a registration page without encryption traffic is unacceptable.

Reply:
I see your point and fully agree regarding providing a registration on an unencrypted channel.

I think i was trying not to assume how people would be doing this, as this might be taken care of differently if running synapse in a Kubernetes environment, for instance. Hence the "Further Resources".